workspaces/mcp-chat-client/plugins/mcp-chat-client/src/components/RightPane/ActiveMcpServers.tsx

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useTheme } from '@mui/material/styles';
+import MemoryIcon from '@mui/icons-material/Memory';
+import Box from '@mui/material/Box';
+import Chip from '@mui/material/Chip';
+import Tooltip from '@mui/material/Tooltip';
+import Typography from '@mui/material/Typography';
+
+interface MCPServer {
+  id?: string;
+  name: string;
+  enabled: boolean;
+  type?: string;
+  hasUrl?: boolean;
+  hasNpxCommand?: boolean;
+  hasScriptPath?: boolean;
+}
+
+interface ActiveMcpServersProps {
+  mcpServers: MCPServer[];
+  onServerToggle: (serverName: string) => void;
+}
+
+export const ActiveMcpServers = ({
+  mcpServers,
+  onServerToggle,
+}: ActiveMcpServersProps) => {
+  const theme = useTheme();
+  const isDarkMode = theme.palette.mode === 'dark';
+
+  // Helper functions to avoid nested ternary expressions
+  const getChipBackgroundColor = (server: MCPServer) => {
+    if (server.enabled) {
+      return isDarkMode ? '#2d4a2d' : '#e8f5e8';
+    }
+    return 'transparent';
+  };
+
+  const getChipColor = (server: MCPServer) => {
```

Comment:
use material-ui theming for all these colors so that the plugin works with custom themes. Same in other files